subsys/bluetooth/mesh/gen_prop_srv.c

```diff
@@ -404,19 +404,17 @@ static void handle_user_property_get(struct bt_mesh_model *mod,
 
 	net_buf_simple_add_u8(&rsp, prop->user_access);
 
-	if (!(prop->user_access & BT_MESH_PROP_ACCESS_READ)) {
-		goto respond;
-	}
-
-	struct bt_mesh_prop_val value = {
-		.meta = *prop,
-		.value = net_buf_simple_tail(&rsp),
-		.size = CONFIG_BT_MESH_PROP_MAXSIZE,
-	};
+	if (prop->user_access & BT_MESH_PROP_ACCESS_READ) {
+		struct bt_mesh_prop_val value = {
```

Comment:
I don't understand what was fixed here, is it just to help Coverity?

Reply:
Coverity is upset about the initialization of variable `value` being bypassed by the goto. The code flow seems at least as readable when we flip the if and thereby reduce the scope, so I decided to just do that instead of silencing the error. It makes no difference to the actual behavior of the code.

Reply:
Is it possible to change the code so the other "goto respond" can be removed too? By adding the !prop test to this test. and moving the net_buf into the if statemet.

Reply:
I mean, it is possible, but that would move the main code flow path into two levels of indentation. I'd rather keep it the way it is